internal/websocket/ws.go

```diff
@@ -134,8 +179,13 @@ func (ws *WebSocket) initialize() {
 					// TODO need to find a proper way to log this error
 					continue
 				}
-
-				ws.respChan <- res
+				responseChan, ok := ws.getResponseChannel(fmt.Sprintf("%v", res.ID))
+				if ok {
```

Comment:
I would suggest inversing the if with `!ok` and getting ride of else doing return first approach